libqtile/widget/redshift.py

```diff
@@ -0,0 +1,333 @@
+        """
+        subprocess.run(["redshift", "-x"], check=True)
+
+
+class Redshift(_TextBox, base.PaddingMixin):
+    """
+    Redshift widget provides the following functionality:
+      - Call redshift with a specific brightness, temperature and gamma config without using your location.
+
+    The redshift command can be called by just left-clicking the widget and disabling it the same way.
+
+    Scrolling through the widget will show the settings,
+    ie. the brightness, gamma and temperature that redshift
+    will use if the widget is enabled.
+    """
+
+    redshift_driver = RedshiftDriver()
+    orientations = base.ORIENTATION_HORIZONTAL
+
+    defaults = [
+        ("brightness", 1.0, "Redshift brightness"),
+        (
+            "disabled_txt",
+            "󱠃",
+            "Text to show when redshift is disabled. NOTE: by default, a nerd icon is used."
+            "Available fields: 'brightness' brightness to set when redshift is enabled, "
+            "'is_enabled' boolean to state whether the widget is enabled or not, "
+            "'gamma_blue' gamma blue value to set when redshift is enabled, "
+            "'gamma_green' gamma green value to set when redshift is enabled, "
+            "'gamma_red' gamma red value to set when redshift is enabled, "
+            "'temperature' temperature to set when redshift is enabled, ",
+        ),
+        (
+            "enabled_txt",
+            "󰛨",
+            "Text to show when redshift is disabled. NOTE: by default, a nerd icon is used."
+            "Available fields: see disabled_txt's available fields for all of them",
+        ),
+        ("gamma_red", 1.0, "Redshift gamma red setting"),
+        ("gamma_blue", 1.0, "Redshift gamma blue"),
+        ("gamma_green", 1.0, "Redshift gamma green"),
+        ("font", "sans", "Default font"),
+        ("fontsize", 20, "Font size"),
+        ("foreground", "ffffff", "Font colour for information text"),
+        ("temperature", 1700, "Redshift temperature to set when enabled"),
+        (
+            "temperature_fmt",
+            "Temperature: {temperature}",
+            "Text to display when showing temperature text"
+            "Available fields: 'temperature' temperature to set when redshift is enabled. ",
+        ),
+        (
+            "brightness_fmt",
+            "Brightness: {brightness}",
+            "Text to display when showing brightness text"
+            "Available fields: 'brightness' brightness to set when redshift is enabled. ",
+        ),
+        (
+            "gamma_fmt",
+            "Gamma: {gamma_red}:{gamma_green}:{gamma_blue}",
+            "Text to display when showing gamma text"
+            "Available fields: 'gamma_red' gamma red value to set when redshift is enabled. "
+            "'gamma_blue' gamma blue value to set when redshift is enabled. "
+            "'gamma_green' gamma green value to set when redshift is enabled. ",
+        ),
+    ]
+
+    _dependencies = ["redshift"]
+
```

Comment:
```suggestion

    supported_backends = {"x11"}

```
My understanding is that `redshift` only works on x11 so we need to make sure it's not loaded in wayland systems.

Reply:
Yep, thanks that explains the failed `wayland` test too. I've now skipped the tests if we are on `wayland`